In [2]:
import h5py

In [3]:
gate_path = 'Results/Unsteady/Output/Output Blocks/Base Output/Unsteady Time Series/SA 2D Area Conn'
file_path = r'W:\2ERDC12 - Clearwater\ClearwaterHECRAS_testCases\from_Andromeda_2024_0116\ClearwaterHECRAS_testCases\HECRAS_testCases_v1\clearWaterTestCases.p52.hdf'

In [4]:
infile = h5py.File(file_path, 'r')

In [6]:
try:
    group = infile[gate_path]
    g = True
except KeyError:
    g = False

print(g)

True


In [7]:
if 'SA 2D Area Conn' in list(infile['Results/Unsteady/Output/Output Blocks/Base Output/Unsteady Time Series/']):
    gates = list(infile['Results/Unsteady/Output/Output Blocks/Base Output/Unsteady Time Series/SA 2D Area Conn'].keys())
    for g in gates:
        f = f'Results/Unsteady/Output/Output Blocks/Base Output/Unsteady Time Series/SA 2D Area Conn/{g}'
        flow = infile[f'{f}/HW TW Segments/Flow'][()]
        headwater_cells = infile[f'{f}/HW TW Segments/Headwater Cells'][()]
        tailwater_cells = infile[f'{f}/HW TW Segments/Tailwater Cells'][()]
        hw_tw_station = infile[f'{f}/HW TW Segments/HW TW Station'][()]
        structure_variables = infile[f'{f}/Structure Variables'][()]
        print(flow.shape)
        print(headwater_cells.shape)
        print(tailwater_cells.shape)
        

(5761, 2)
(1,)
(1,)


In [10]:
import numpy as np
stacked = np.stack((headwater_cells, tailwater_cells))
stacked

array([[b'       126'],
       [b'       114']], dtype='|S10')

In [ ]:
gate_connectivity = np.stack((headwater_cells[()], tailwater_cells[()]), axis=1)
gate_connectivity

array([[b'       126', b'       114']], dtype='|S10')

In [11]:
import numpy as np
concatenated = np.concatenate((headwater_cells, tailwater_cells))
concatenated 

array([b'       126', b'       114'], dtype='|S10')

Confirm all flow values are stored in the first "connection".

In [48]:
print(flow[:,0].sum())
print(flow[:,0].min())
print(flow[:,0].max())

-203690.38
-39.999416
0.5331346


In [16]:
flow[:,-1].shape

(5761,)

In [49]:
print(flow[:,1].sum())
print(flow[:,1].min())
print(flow[:,1].max())

0.0
0.0
0.0


In [51]:
print(headwater_cells)
print(tailwater_cells)

[b'       126']
[b'       114']


In [66]:
hw_tw_station.shape

(2,)

So, it looks like water is flowing from cell 126 to 114 and that information is stored in the 0 column.

In [64]:
structure_variables[:,0].sum() == flow[:,0].sum()

True

Confirmed above that the sum of the first cell equals the sum of trhe first column of structure variables.